packages/lodestar/src/network/util.ts

```diff
@@ -133,14 +133,16 @@ export async function dialProtocol(
   libp2p: LibP2p,
   peerId: PeerId,
   protocol: string,
-  timeout: number
+  timeout: number,
+  signal: AbortSignal | undefined
 ): ReturnType<LibP2p["dialProtocol"]> {
   const abortController = new AbortController();
   const timer = setTimeout(() => {
     abortController.abort();
   }, timeout);
+  signal?.addEventListener("abort", () => clearTimeout(timer), {once: true});
   try {
-    const conn = await libp2p.dialProtocol(peerId, protocol, {signal: abortController.signal} as object);
+    const conn = await libp2p.dialProtocol(peerId, protocol, {signal} as object);
```

Comment:
I think we want `abortController.signal` as original

Reply:
ah yeah, good catch

Reply:
reverted